model-client/src/jvmMain/kotlin/org/modelix/model/client2/LazyLoading.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.client2
+
+import kotlinx.coroutines.runBlocking
+import org.modelix.model.IKeyListener
+import org.modelix.model.IKeyValueStore
+import org.modelix.model.IVersion
+import org.modelix.model.lazy.BranchReference
+import org.modelix.model.lazy.CLVersion
+import org.modelix.model.lazy.CacheConfiguration
+import org.modelix.model.lazy.ObjectStoreCache
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.persistent.HashUtil
+
+/**
+ * IModelClientV2#loadVersion eagerly loads the whole model. For large models this can be slow and requires lots of
+ * memory.
+ * This function loads parts of the model lazily while it is iterated and limits the amount of data that is cached on
```

Comment:
> By convention, the first paragraph of the documentation text (the block of text until the first blank line) is the summary description of the element, and the following text is the detailed description.

https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax

I think the second sentence would make a good summary for the first paragraph.

Reply:
Changed the order.

Reply:
Would also required a newline after the sentence to fulfill the convention.

Reply:
Added an empty line.